service.go

```diff
@@ -328,7 +328,7 @@ func (s *Service) Start(useRegistry bool, profile string, confDir string) (err e
 		go processAsyncResults()
 	}
 
-	err = s.proto.Initialize(s.lc, s.asyncCh)
+	err = s.proto.Initialize(s, s.lc, s.asyncCh)
```

Comment:
Maybe `s.lc` and `s.asyncCh` can be removed, as we have `s`

Reply:
@jduranf Thanks for the review and comments. I agree this could be simplified, but we'd have to change the **Service** struct too, as **lc** and **asyncCh** are both private, so concrete types implementing **ProtocolDriver** wouldn't be able to access either.

I'd actually been considering removing all of the clients from the **Service** struct and making them top-level globals in the device package.

We could also add a top-level `(s *Service) LoggingClient()` method which a **ProtocolDriver** implementation could use to access the logging client, and then get rid of the 2nd argument here.

I'm going to go ahead and merge 'as is'.

Unless there are any objections, why don't you go ahead and create a PR to add a method to return the logging client as suggested above, and then simplify the signature of Initialize in **ProtocolDriver**?